pkg/util/rolloutblock/rolloutblock.go

```diff
@@ -0,0 +1,58 @@
+package rolloutblock
```

Comment:
Taking into account a massive split-concat pattern in your code I might suggest introducing an additional level of abstraction, like: RolloutBlockSet, which is effectively an ordered set of rollout blocks. It can encapsulate all diff, includes, excludes etc logic and simplify strings operations in the modules above.